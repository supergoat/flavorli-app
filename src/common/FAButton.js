// @flow
import React, { Component } from "react";
import {
  Text,
  TouchableOpacity
} from "react-native";
import type {
   ViewStyleProp
} from 'react-native/Libraries/StyleSheet/StyleSheet';
import FAColors from "../common/FAColors";
import FAStyleSheet from "../common/FAStyleSheet";

/** ============================================================================
<FAButton
  type={'primary' | 'secondary'}
  text={string}
  style={ViewStyleProp}
/>
--------------------------------------------------------------------------------
A flavorli themed button
============================================================================= */
type Props = {
  type: 'primary' | 'secondary',
  text: string,
  style?: ViewStyleProp
};
class FAButton extends Component<Props> {
  render() {
    const { type, text, style, ...rest } = this.props;

    return (
      <TouchableOpacity
        {...rest}
        activeOpacity={type === 'primary' ? 0.8 : 0.6}
        style={[
          styles.button,
          type === 'primary' ? styles.primary : styles.secondary,
          style
        ]}
      >
        <Text style={[
          styles.buttonText,
          type === 'secondary' && styles.secondaryButtonText,
        ]}>
          {text}
        </Text>
      </TouchableOpacity>
    );
  }
}

/* StyleSheet
============================================================================= */
const styles = FAStyleSheet.create({
  button: {
    flex: 1,
    padding: 15,
    borderRadius: 3,
    borderWidth: 1
  },
  primary: {
    backgroundColor: FAColors.oxfordBlue,
  },
  secondary: {
    borderColor: FAColors.oxfordBlue,
  },
  buttonText: {
    color: FAColors.white,
    fontFamily: 'Lato'
  },
  secondaryButtonText: {
    color: FAColors.oxfordBlue
  }
});

/* Export
============================================================================= */
export default FAButton;
