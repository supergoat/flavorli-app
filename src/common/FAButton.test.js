// @flow
import React from 'react';
import { View, Text } from 'react-native';
import FAButton from './FAButton';
import renderer from 'react-test-renderer';

describe('FAButton', () => {
  it('renders correctly when the button is of type primary', () => {
    const tree = renderer.create(
      <FAButton
        type='primary'
        text='button'
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when the button is of type secondary', () => {
    const tree = renderer.create(
      <FAButton
        type='secondary'
        text='button'
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
