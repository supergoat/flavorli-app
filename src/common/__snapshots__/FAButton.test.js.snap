// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAButton renders correctly when the button is of type primary 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#333C4E",
      "borderRadius": 3,
      "borderWidth": 1,
      "flex": 1,
      "opacity": 1,
      "padding": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Lato",
        },
        false,
      ]
    }
  >
    button
  </Text>
</View>
`;

exports[`FAButton renders correctly when the button is of type secondary 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "#333C4E",
      "borderRadius": 3,
      "borderWidth": 1,
      "flex": 1,
      "opacity": 1,
      "padding": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Lato",
        },
        Object {
          "color": "#333C4E",
        },
      ]
    }
  >
    button
  </Text>
</View>
`;
